src/app/component/eco-news/services/eco-news.service.ts

```diff
@@ -52,4 +61,9 @@ export class EcoNewsService {
   public getRecommendedNews(id: number): Observable<EcoNewsModel> {
     return this.http.get<EcoNewsModel>(`${this.backEnd}econews/recommended?openedEcoNewsId=${id}`);
   }
+
+  ngOnDestroy(): void {
+    this.destroy.next(null);
```

Comment:
why did you use nulll?

Reply:
I found this construction to avoid subscriptions: https://medium.com/ngx/why-do-you-need-unsubscribe-ee0c62b5d21f

Reply:
Use takeUntill